services/spar/src/Spar/Orphans.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE RecordWildCards       #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE StandaloneDeriving    #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeSynonymInstances  #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Spar.Orphans where
+
+import Data.Aeson
+import Data.CaseInsensitive
+import Data.Id
+import Data.String.Conversions
+import GHC.Generics
+import Servant hiding (URI)
+import Spar.Types
+
+
+instance FromHttpApiData UserId where
+  parseUrlPiece = fmap Id . parseUrlPiece
+
+instance ToHttpApiData UserId where
+  toUrlPiece = toUrlPiece . show
+
+deriving instance Generic ServantErr
+instance FromJSON ServantErr
+instance ToJSON ServantErr
+
+instance MimeRender PlainText Void where
+  mimeRender _ = error "instance MimeRender HTML Void: impossible"
+
+instance FromJSON (CI SBS) where parseJSON = fmap mk . parseJSON
+instance ToJSON (CI SBS)   where toJSON = toJSON . original
+
+instance FromJSON SBS      where parseJSON = fmap (cs @ST @SBS) . parseJSON
+instance ToJSON SBS        where toJSON = toJSON . cs @SBS @ST
```

Comment:
These instances only work when you have UTF-8-encoded strings. If it's necessary to have them around, at least we can base64-encode the bytestrings. If this is impossible, and having `Text` instead of `ByteString` inside of the relevant datatypes is impossible either, then I'd rather have manually written instances – this way nobody will get bitten.

Reply:
turns out they weren't needed anywhere...?  removed, thanks!